main.tf

```diff
@@ -99,6 +99,10 @@ module "budget" {
   monitoring_notification_channels = var.budget_monitoring_notification_channels
   display_name                     = var.budget_display_name != null ? var.budget_display_name : null
   labels                           = var.budget_labels
+  calendar_period                  = var.calendar_period
```

Comment:
nit: lets prefix these vars used in root module with `budget_`, so `budget_calendar_period` etc